trustfall_core/src/ir/mod.rs

```diff
@@ -636,7 +623,7 @@ impl<LeftT: NamedTypedValue> Operation<LeftT, Argument> {
                 // For the operands relative to each other, nullability doesn't matter,
                 // but the rest of the type must be the same.
                 let right_type = right_type.unwrap();
-                if are_base_types_equal_ignoring_nullability(&left_type.base, &right_type.base) {
+                if are_base_types_equal_ignoring_nullability(left_type, right_type) {
```

Comment:
This function and similar ones below might be good to refactor and possibly either expose as methods directly, or reimplement using appropriate helper methods.

They were free functions because we couldn't implement methods on a foreign `Type` type. Now the type is ours so we can do better.